products/platform/product-slo-tools/README.md

```diff
@@ -0,0 +1,157 @@
+            thresholds: 99.9 # thresholds for reporting/alerting
+        - latency:
+            type: latency
+            numerator: 'promql-goes-here' # number of "good requests"
+            denominator: 'promql-goes-here' # number of "all requests expected to be good"
+            thresholds:
+              p99: 1000
+              p90: 500
+
+  ```
+
+
+VFS team perspective:
+
+- meet with VSP to do introduction of SLO requirement(s)
+- define initial SLO based on engineer & stakeholder perspective of the product's optimal user experience
+- get VSP and VFS stakeholder approval and consensus on practicality of proposed SLO definition
+- translate into SLO configuration (being generous with initial thresholds)
+- open PR(s) to "ship" the SLO into production
+- continually monitor based on alerting rules
+- follow protocol in collaboration cycle that engineering & product effort must be driven based on SLO performance
+
+
+## Assumptions
+
+- We define a service or a product as a discrete block of functionality on VA.gov
+- A `product` is some code that's compiled & released on networked infra that users access via the web
+- Teams (VFS) may have responsibility for one-to-many products, each having one-to-many SLOs
+
+Maturity level of a given product could be:
+
+- greenfield - with no current deploys on the VSP
+- existing system in production, with some form of monitoring, yet no formal objectives exist, thus no error budget and an implicit 100% uptime goal
+- running deployments in production, with SLOs below 100%, yet with no common understanding on how to leverage SLO - "without teeth"
+
+### Risks
+
+- We don't know exactly what makes a useful product/service boundary from a code perspective which might make defining
+    ownership difficult
+- SLO compliance may be simply another KPI or metric that's not used as a decision-making tool (we need to build policy
+    around this?)
+- Increased confusion around how to configure and define applications, this may slow delivery
+- VFS teams may not be able to improve/meet SLO because of external factors (partner services down, etc).
+
+### Alternatives Considered
+
+#### Define products and SLOs in [Datadog](https://www.datadoghq.com/blog/slo-monitoring-tracking/)
+
+Datadog provides a nice set of tools for defining & monitoring SLOs, we could use this for the solution.
+
+- This approach requires Datadog to be fully operational in our infrastructure
+- This approach requires license & volume pricing for all teams, and costs scale per node + team
+
+#### Build a dashboard manually in Grafana for each product
+
+We could build a distinct SLO dashboard in Grafana for each product.
+
+- This approach is slow and laborious
+- Would require training VFS teams on Grafana & PromQL (or future tech)
+- Tightly couples SLO configuration to dashboarding solution
+- Data integrity and configuration requires updating & exporting from Grafana directly
```

Comment:
Would it be possible to have a single dashboard with some kind of big dropdown list of endpoints? ...I suppose the actual thresholds would also need to be some kind of dropdown 🤔

Reply:
Yeah, I'm looking into Grafana 7.0 features that allow for variable thresholds